admin/services/jwt-service.js

```diff
@@ -0,0 +1,73 @@
+'use strict'
+const Promise = require('bluebird')
+const crypto = Promise.promisifyAll(require('crypto'))
+const jwt = Promise.promisifyAll(require('jsonwebtoken'))
+const uuidv4 = require('uuid/v4')
+const ObjectId = require('mongoose').Types.ObjectId
+
+const Pupil = require('../models/pupil')
+
+/** @namespace */
+
+const jwtService = {
+  /**
+   *
+   * @param pupil
+   * @return {*}
+   */
+  createToken: async (pupil) => {
+    if (!(pupil && pupil._id)) {
+      throw new Error('Pupil is required')
+    }
+    const jwtId = uuidv4()
+    pupil.jwtSecret = await crypto.randomBytes(32).toString('hex')
+    await pupil.save()
+
+    // TODO: for additional security add in a device Id
+    const payload = {
+      iss: 'MTC Admin',                                       // Issuer
+      sub: pupil._id,                                         // Subject
+      exp: Math.floor(Date.now() / 1000) + (60 * 60),         // Expiry
+      nbf: Math.floor(Date.now() / 1000),                     // Not before
+      jwi: jwtId                                              // JWT token ID
+    }
+
+    // Construct a JWT token
+    const token = await jwt.sign(payload, pupil.jwtSecret)
+    return token
```

Comment:
you don't need the variable here you can just return await jwt.sign(...

Reply:
true - but I _like_ it this way.  The compiler will handle it.